server/templates/staff/course/assignment/assignment.new.html

```diff
@@ -25,33 +25,74 @@
-                  {{ forms.render_field(form.autograding_key, label_visible=true, placeholder='5abc... (Optional)', type='text') }}
-                  {{ forms.render_checkbox_field(form.revisions_allowed, label_visible=true) }}
-                  {{ forms.render_checkbox_field(form.uploads_enabled, label_visible=true) }}
-                  {{ forms.render_field(form.upload_info, label_visible=true, placeholder='Upload the following files: test.py', type='text') }}
+                </fieldset>
+                <h3>Deadlines</h3>
+                <fieldset>
+                    {{ forms.render_field(form.due_date, label_visible=true, value=utils.new_due_date(current_course), type='text', class='form-control datepicker') }}
+                    {{ forms.render_field(form.lock_date, label_visible=true, value=utils.new_lock_date(current_course), type='text', class='form-control datepicker') }}
+                </fieldset>
+                <h3>Grading</h3>
+                <fieldset>
+                    {{ forms.render_checkbox_field(form.visible, label_visible=true) }}
+                    {{ forms.render_field(form.autograding_key, label_visible=true, placeholder='5abc... (Optional)', type='text') }}
+                    {{ forms.render_checkbox_field(form.revisions_allowed, label_visible=true) }}
+                    {{ forms.render_checkbox_field(form.uploads_enabled, label_visible=true) }}
+                    {{ forms.render_field(form.upload_info, label_visible=true, placeholder='Upload the following files: test.py', type='text') }}
+                </fieldset>
               {% endcall %}
             </div>
           </div>
           <!-- /.box-body -->
         </div>
         <!-- /.box -->
       </div>
-
-
-
-
-
   <!-- </body> do not close body in template-->
 {% endblock %}
+
+{% block page_js %}
+<script type="text/javascript"  src="https://cdnjs.cloudflare.com/ajax/libs/jquery-steps/1.1.0/jquery.steps.js"></script>
+<script type="text/javascript"  src="https://cdnjs.cloudflare.com/ajax/libs/jquery-validate/1.15.1/jquery.validate.min.js"></script>
+<script>
+var form = $(".new-assign-form").show();
+form.steps({
+    headerTag: "h3",
+    bodyTag: "fieldset",
+    transitionEffect: "slideLeft",
+    onStepChanging: function (event, currentIndex, newIndex) {
+        // Allways allow previous action even if the current form is not valid!
+        if (currentIndex > newIndex)
+        {
+            return true;
+        }
+        form.validate().settings.ignore = ":disabled,:hidden";
+        return form.valid();
+    },
+    onFinished: function (event, currentIndex) {
+        form.submit();
+    }
+})
+var nameTweaked = false;
+$('#display-name').on('change, keyup', debounce(function () {
+    if (!nameTweaked) {
+        display_name = $('#display-name').val().toLowerCase().split(' ').join('')
+        course_offering = $.trim($("#course-offering").text())
```

Comment:
It is cleaner but it constrains this to be in the template's `page_js` forever. Maybe I can make this a function and then use jinja. 

Another option is to do it once and use the value of the assignment-name field.